### ISSUE TITLE
File descriptor leaks when used in nginx-lua-module

### ISSUE THREAD

**Issue by jonas0616** [agladysh/lua-geoip#22]
When `lua_code_cache` is `off`, fd which points to error.log increases continuously for workers.
I do understand `lua_code_cache off` should not be used in production. This setting exists in development environment only. However, it still annoys because nginx will no longer accept any request when fd limit is reached during testing.

```
$ ll /proc/{worker_pid}/fd/
l-wx------ 1 ubuntu ubuntu 64 May  8 06:43 14 -> /var/log/nginx/error.log
l-wx------ 1 ubuntu ubuntu 64 May  8 06:43 2 -> /var/log/nginx/error.log
l-wx------ 1 ubuntu ubuntu 64 May  8 06:43 28 -> /var/log/nginx/error.log
l-wx------ 1 ubuntu ubuntu 64 May  8 06:43 30 -> /var/log/nginx/error.log
l-wx------ 1 ubuntu ubuntu 64 May  8 06:43 33 -> /var/log/nginx/error.log
l-wx------ 1 ubuntu ubuntu 64 May  8 06:43 35 -> /var/log/nginx/error.log
l-wx------ 1 ubuntu ubuntu 64 May  8 06:43 37 -> /var/log/nginx/error.log
l-wx------ 1 ubuntu ubuntu 64 May  8 06:43 39 -> /var/log/nginx/error.log
l-wx------ 1 ubuntu ubuntu 64 May  8 06:43 42 -> /var/log/nginx/error.log
l-wx------ 1 ubuntu ubuntu 64 May  8 06:43 44 -> /var/log/nginx/error.log
...
```

After some experiments and observation, I can reproduce by the following simplified scripts:
Entry script:
```
#!/usr/bin/env lua
local Geo = require "MyGeo"
```

MyGeo.lua:
```
#!/usr/bin/env lua
local geoip_country = require "geoip.country"

local GeoIp = {
    db_country = geoip_country.open("/usr/share/GeoIP/GeoIP.dat")
}

return GeoIp
```

The geoip module is 
```
# luarocks show lua-geoip

lua-geoip 0.1.2-1 - Bindings for MaxMind's GeoIP library

License:        MIT/X11
Homepage:       http://github.com/agladysh/lua-geoip
Installed in:   /usr/local

Modules:
        geoip.country
        geoip
        geoip.city
```

The openresty version is:
```
./nginx -v
nginx version: openresty/1.11.2.1
```

**Comment by agladysh** [comment]
Thank you for the bug report. Looks like #5 should have fixed that. Can you please check if the master HEAD has the same issue?

**Comment by jonas0616** [comment]
@agladysh I see. This pull request has already been merged since two years ago. Any chance to release with this bug fix and make luarocks repository updated?

**Comment by agladysh** [comment]
Please confirm that the master HEAD fixes it, and I will make release this week.

**Comment by jonas0616** [comment]
@agladysh I just build with master HEAD and it seems that bug is fixed.

By the way, I have to rebuild nginx (openresty) with option `--with-http_geoip_module`, or it throws exception `undefined symbol: GeoIP_continent_by_id`. I have no idea it should blames to **self-build** instead of luarocks version, or there is some changes in lua-geoip.

**Comment by agladysh** [comment]
Many thanks for checking, I will do a release this week.

Not sure about the build problem, sorry.

**Comment by agladysh** [comment]
@jonas0616 I released v0.2.
